consensus/exit_probability_randomizer/exit_prob_map.h

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2018 IOTA Stiftung
+ * https://github.com/iotaledger/entangled
+ *
+ * Refer to the LICENSE file for licensing information
+ */
+
+#ifndef __CONSENSUS_EXIT_PROBABILITY_RANDOMIZER_EXIT_PROB_MAP_H__
+#define __CONSENSUS_EXIT_PROBABILITY_RANDOMIZER_EXIT_PROB_MAP_H__
+
+#include "common/errors.h"
+#include "consensus/cw_rating_calculator/cw_rating_calculator.h"
+#include "consensus/exit_probability_randomizer/exit_probability_randomizer.h"
+#include "consensus/exit_probability_validator/exit_probability_validator.h"
+#include "utils/containers/hash/hash_double_map.h"
+#include "utils/time.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct ep_prob_map_randomizer_s ep_prob_map_randomizer_t;
+struct ep_prob_map_randomizer_s {
+  ep_randomizer_t base;
+  hash_to_double_map_t exit_probs;
+  hash_to_double_map_t transition_probs;
+};
+
+void iota_consensus_exit_prob_map_reset(
+    ep_randomizer_t *const exit_probability_randomizer);
+
+void iota_consensus_exit_prob_map_init(ep_randomizer_t *const randomizer);
+
+/**
+ * Randomize a tip selection based on exit probabilities
```

Comment:
Could you also add who needs to own a lock if any to these comments?

Reply:
Added doc, but it does make sense to add a lock to the class